dali/dfu/dfurun.cpp

```diff
@@ -1355,23 +1341,32 @@ class CDFUengine: public CInterface, implements IDFUengine
                         }
 
                         bool dirPerPart = false;
-                        //MORE: This could be combined with the code that gets defaultSprayParts
-                        if (isContainerized())
+                        IDFUfileSpec *dst = wu->queryUpdateDestination();
+                        // keys default wrap for copy
+                        if (destination->getWrap()||(iskey&&(cmd==DFUcmd_copy)))
+                        {
+                            dst->setNumPartsOverride(srcFile->numParts());
+                        }
+                        else if (isContainerized())
                         {
                             StringBuffer clusterName;
                             destination->getGroupName(0, clusterName);
                             Owned<IPropertyTree> plane = getStoragePlane(clusterName);
                             if (plane)
                             {
-                                dirPerPart = plane->getPropBool("@subDirPerFilePart", true);
+                                if (plane->hasProp("@defaultSprayParts"))
+                                {
+                                    if (destination->getNumPartsOverride()==0) // use destination defaultSprayParts if requestor doesn't provide num parts
+                                        dst->setNumPartsOverride(plane->getPropInt("@defaultSprayParts"));
+                                    dirPerPart = plane->getPropBool("@subDirPerFilePart", true);
```

Comment:
this is now inside the if (plane->hasProp("@defaultSprayParts")) block..

As before the change, it should be read regardless of whether the plane has @defaultSprayParts set,
i.e. should be moved out of the block.